pkgs/applications/audio/carla/common.nix

```diff
@@ -0,0 +1,109 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, alsa-lib
+, file
+, fluidsynth
+, jack2
+, liblo
+, libpulseaudio
+, libsndfile
+, pkg-config
+, python3Packages
+, which
+, gtk2 ? null
+, gtk3 ? null
+, qtbase ? null
+, withFrontend ? true
+, withGtk2 ? true
+, withGtk3 ? true
+, withQt ? true
+, wrapQtAppsHook ? null
+, version
+, rev
+, hash
+}:
+
+assert withFrontend -> python3Packages ? pyqt5;
+assert withQt -> qtbase != null;
+assert withQt -> wrapQtAppsHook != null;
+assert withGtk2 -> gtk2 != null;
+assert withGtk3 -> gtk3 != null;
+
+stdenv.mkDerivation rec {
+  pname = "carla";
+  inherit version rev hash;
+
+  src = fetchFromGitHub {
+    owner = "falktx";
+    repo = pname;
+    rev = rev;
+    hash = hash;
+  };
+
+  nativeBuildInputs = [
+    python3Packages.wrapPython pkg-config which wrapQtAppsHook
+  ];
+
+  pythonPath = with python3Packages; [
+    rdflib pyliblo
+  ] ++ lib.optional withFrontend pyqt5;
+
+  buildInputs = [
+    file liblo alsa-lib fluidsynth jack2 libpulseaudio libsndfile
+  ] ++ lib.optional withQt qtbase
+    ++ lib.optional withGtk2 gtk2
```

Comment:
We should think about removing gtk2

Reply:
IIRC, gtk2 is needed for plugins whose interface is implemented using gtk2. I can't think of an example plugin right now, but removing it might break the setup of some people using gtk2-based audio plugins